File: BusinessLogic/NamedBO.py
from BusinessLogic import BusinessObject


class NamedBO(BusinessObject):
    def __init__(self):
        super().__init__()
        self._erstellungs_zeitpunkt = ""
        self._name = ""

    def set_erstellungs_zeitraum(self, erstellungs_zeitraum):
        self._erstellungs_zeitpunkt = erstellungs_zeitraum

    def get_erstellungs_zeitraum(self):
        return self._erstellungs_zeitpunkt

    def set_name(self, name):
        self._name = name

    def get_name(self):
        return self._name

File: frontend/src/api/NamedBO.js
import BusinessObject from './BusinessObject';

export default class NamedBO extends BusinessObject {

/** Ein Name kann im Constructor von den Subklassen angenommen
     und hier in das Attribut name geschrieben werden. */
   constructor(name) {
       super();
        this.erstellungs_zeitpunkt = null;
        this.name = name;
   }

    setErstellungsZeitpunkt(erstellungs_zeitpunkt) {
        this.erstellungs_zeitpunkt = erstellungs_zeitpunkt
    }

    getErstellungsZeitpunkt() {
        return this.erstellungs_zeitpunkt
    }

    setName(name) {
        this.name = name
    }

    getName() {
        return this.name
    }
}